app/src/main/java/io/apicurio/registry/services/auth/CustomAuthenticationMechanism.java

```diff
@@ -115,6 +118,8 @@ public void init() {
         }
     }
 
+    @Retry(abortOn = {NotAuthorizedException.class, ForbiddenException.class}) // 3 retries, 200ms jitter
+    @Timeout(3000) // 3000ms
     public Uni<SecurityIdentity> customAuthentication(RoutingContext context, IdentityProviderManager identityProviderManager) {
```

Comment:
Is this where the Retry should happen?  Or should we be more targeted by wrapping the interactions with external systems in CDI beans so we can put the fault tolerance at a more granular location?  Honest question because I'm not sure.

Which lines of code in this method might fail and need to be retried?  It's definitely line 125/130 (OidcAuth).  Is it also 121/126 and 129/134 and 133/138?  It's hard to tell.

Reply:
Agreed, I think we should extract some parts of the code here so we can retry at a more granular level. There are some situations in this method that are worth a retry () but others, IMO, don't.